src/components/CertificateSettings.vue

```diff
@@ -31,31 +32,43 @@
 			@input="savedCertificate = null" />
 		<NcSelect v-if="alias !== null"
 			v-model="savedCertificate"
+			class="certificate-settings__certificate"
 			:options="smimeCertOptions"
 			:aria-label-combobox="t('mail', 'Select certificates')"
 			:searchable="false" />
-		<ButtonVue type="primary"
+		<NcButton type="primary"
+			class="certificate-settings__submit"
 			:disabled="certificate === null"
 			:aria-label="t('mail', 'Update Certificate')"
 			@click="updateSmimeCertificate">
 			{{ t('mail', 'Update Certificate') }}
-		</ButtonVue>
+		</NcButton>
+		<NcNoteCard v-if="alias && !savedCertificate.isChainVerified"
+			type="warning">
+			<p>{{ t('mail', 'The selected certificate is not trusted by the server. Recipients might not be able to verify your signature.') }}</p>
+		</NcNoteCard>
 	</div>
 </template>
 
 <script>
-import { NcSelect, NcButton as ButtonVue } from '@nextcloud/vue'
+import { NcSelect, NcButton, NcNoteCard } from '@nextcloud/vue'
 import { compareSmimeCertificates } from '../util/smime.js'
 import { mapGetters } from 'vuex'
 import { showError, showSuccess } from '@nextcloud/dialogs'
 import Logger from '../logger.js'
 import moment from '@nextcloud/moment'
 
+const noCertificateOption = () => ({
```

Comment:
what's the purpose of the function? could this be inlined?

Reply:
I moved it out to prevent duplication but it can be part of the component as well. The default option is created in multiple places inside the component.

I'll move it back in the component.